docs/user-journeys/users/application-developer/foundational.md

```diff
@@ -9,80 +9,151 @@ track: "USERS › APPLICATION DEVELOPER › FOUNDATIONAL"
+
+* **{% glossary_tooltip text="PersistentVolumes" term_id="persistent-volume" %}** and **{% glossary_tooltip text="PersistentVolumeClaims" term_id="persistent-volume-claim" %}** - Let you define storage at the cluster level. Typically a cluster administrator defines the PersistentVolume objects for the cluster, and cluster users (application developers, you) define the PersistentVolumeClaim objects that your application requires. Learn [how to set up persistent storage for your cluster](/docs/tasks/configure-pod-container/configure-persistent-volume-storage/) or [read more about persistent volumes](/docs/concepts/storage/persistent-volumes/).
 
-Note that ingress provides networking between your cluster and the outside world. A production cluster should also configure networking between containers and between nodes, which is [defined in a network policy](/docs/tasks/administer-cluster/declare-network-policy/).
+#### Configuration
 
-## How Kubernetes works: an introduction
+To provide configuration data for your application, you can specify Kubernetes environment variables as part of the container definition in your Deployment manifest. However, this may require you to rebuild your container or modify your Deployment.
 
-To work with Kubernetes, you describe your cluster's desired state in terms of Kubernetes API objects. Cluster state includes but is not limited to the following information:
+To decouple configuration data from your workload, you can also use the following objects:
+
+* **{% glossary_tooltip text="ConfigMaps" term_id="configmap" %}** - For non-confidential data, such as an nginx-specific configuration file.
+
+* **{% glossary_tooltip text="Secrets" term_id="secret" %}** - Similar to ConfigMaps, but for confidential data.
+
+*ConfigMaps and Secrets are not overlapping options.* If you have *any* data that you want to keep private, you should be using a Secret. Otherwise there is nothing stopping that data from being exposed to malicious users.
+
+## Understand Kubernetes basics
+
+As an app developer, you don't need to know everything about the inner workings of Kubernetes, but you may find it helpful to understand it at a high-level.
+
+#### Overview
+
+Say that your team is deploying an ordinary Rails application. You've run some calculations and determined that you need five instances of your app running at any given time, in order to handle external traffic.
+
+If you're not running Kubernetes or a similar automated system, you might find the following scenario familiar:
+1. One instance of your app (a complete machine instance or just a container) goes down.
+2. Because your team has monitoring set up, this pages the person on-call.
+3. The on-call has to go in, investigate, and manually spin up a new instance.
```

Comment:
pages the person on call. (no hyphen)
The on-call person has to go in,
the on-call person may also need to ...